File: test_firebase.py
# test_firebase.py
import firebase_admin
from firebase_admin import credentials, firestore

try:
    cred = credentials.Certificate('firebase_credentials.json')
    firebase_admin.initialize_app(cred)
    
    db = firestore.client()
    
    # Test read
    docs = db.collection('bloodStock').get()
    print(f"✅ Python Firebase connected! Found {len(docs)} blood types")
    
except Exception as e:
    print(f"❌ Python connection failed: {e}")

File: test_firebase.js
import { db } from './firebase-config.js';
import { collection, getDocs } from 'https://www.gstatic.com/firebasejs/10.7.1/firebase-firestore.js';

async function testConnection() {
    try {
        const bloodStock = await getDocs(collection(db, 'bloodStock'));
        console.log('✅ Firebase connected!');
        console.log('Blood stock records:', bloodStock.size);
    } catch (error) {
        console.log('❌ Firebase connection failed:', error);
    }
}

testConnection();